app/src/main/java/ch/sdp/vibester/activity/QrScanningActivity.kt

```diff
@@ -0,0 +1,208 @@
+            override fun surfaceCreated(holder: SurfaceHolder) {
+                try {
+                    camera.start(holder)
+                } catch (e: IOException) {
+                    Log.e(getString(R.string.log_tag), e.stackTrace.toString())
+                }
+            }
+
+            @SuppressLint("MissingPermission")
+            override fun surfaceChanged(
+                holder: SurfaceHolder,
+                format: Int,
+                width: Int,
+                height: Int
+            ) {
+                try {
+                    camera.start(holder)
+                } catch (e: IOException) {
+                    Log.e(getString(R.string.log_tag), e.stackTrace.toString())
+                }
+            }
+
+            override fun surfaceDestroyed(holder: SurfaceHolder) {
+                camera.stop()
+            }
+        })
+
+        barcodeDetector.setProcessor(object : Detector.Processor<Barcode> {
+            override fun release() {
+                Toast.makeText(applicationContext, "Scanner has been closed", Toast.LENGTH_SHORT).show()
+            }
+
+            override fun receiveDetections(detections: Detector.Detections<Barcode>) {
+                solveDetection(detections)
+            }
+        })
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        // avoid error using non initialized property
+        if(::camera.isInitialized) {
+            camera.stop()
+        }
+    }
+
+    /*
+     * This section is related to permission request
+     */
+
+    /**
+     * Ask for camera permission
+     */
+    private fun askForCameraPermission() {
+        ActivityCompat.requestPermissions(
+            this@QrScanningActivity,
+            arrayOf(android.Manifest.permission.CAMERA),
+            requestCodeCameraPermission
+        )
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == requestCodeCameraPermission && grantResults.isNotEmpty()) {
+            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                // If permission are granted restart the activity to make sure they are taken into account
+                val intentNew =  Intent(this, QrScanningActivity::class.java)
+                intentNew.putExtra("uidList", uidList)
+                startActivity(intentNew)
+                finish()
+            } else {
+                // Camera permission not granted, come back to previous activity
+                Toast.makeText(applicationContext, "Camera Permission should be granted for that feature", Toast.LENGTH_LONG).show()
+                val returnInt = Intent(this@QrScanningActivity, SearchUserActivity::class.java)
+                startActivity(returnInt)
+                finish()
```

Comment:
You have end and start activity in the beginng of the file too. Maybe create a function fo that